test/language/expressions/class/super-evaluation-order.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: prod-SuperCall
+description: SuperCall should evaluate Arguments prior to checking IsConstructable
+info: |
+  GetSuperConstructor ( )
+
+  [...]
+  1. Let _func_ be ! GetSuperConstructor().
+  2. Let _argList_ be ? ArgumentListEvaluation of |Arguments|.
+  3. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
+  [...]
+features: [default-parameters]
+---*/
+
+var i;
+class A extends {} {
+  constructor() {
+    super(i = 123);
```

Comment:
nice catch!

Reply:
I'm confused ... let me check the specs I'll brb...

Reply:
@leobalter in current spec this test fails, it is for a normative PR from top comment

Reply:
```
Runtime Semantics: ClassDefinitionEvaluation

ClassTail :ClassHeritage { ClassBody }

5. If ClassHeritageopt is not present, then
  ...
6. Else,
  e. If superclass is null, then
    i. Let protoParent be null.
    ii. Let constructorParent be %Function.prototype%.
  f. Else if IsConstructor(superclass) is false, throw a TypeError exception.
```

the current syntax has a dead end.

I believe you need something like a `class A extends null {` here...

Reply:
> @leobalter in current spec this test fails, it is for a normative PR from top comment

is there any other spec PR allowing a class to extend from a non constructor object like `{}`?

I might really be not up to date and missing it. For the current linked change, the coverage here aims for a post evaluated class definition.

Reply:
looking at your snippet you might be right. changing to `null` should work

Reply:
I've updated to extend from a throwing function, so that if `extends null` is ever fixed, this test won't break.